central/image/service/http_handler.go

```diff
@@ -0,0 +1,97 @@
+package service
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+
+	"github.com/pkg/errors"
+	"github.com/stackrox/rox/central/enrichment"
+	"github.com/stackrox/rox/generated/storage"
+	"github.com/stackrox/rox/pkg/httputil"
+	"github.com/stackrox/rox/pkg/images/enricher"
+	"github.com/stackrox/rox/pkg/images/integration"
+	"github.com/stackrox/rox/pkg/scanners/types"
+	"golang.org/x/net/context"
+	"google.golang.org/grpc/codes"
+)
+
+type sbomHandler struct {
+	integration integration.Set
+	enricher    enricher.ImageEnricher
+}
+
+type SBOMRequestBody struct {
+	ClusterID string `json:"clusterID"`
+	ImageName string `json:"imageName"`
+	force     bool   `json:"force"`
+}
+
+func (s *sbomHandler) ImageHandler() http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		if r.Method != http.MethodPost {
+			w.WriteHeader(http.StatusMethodNotAllowed)
+			return
+		}
+		var params SBOMRequestBody
+		err := json.NewDecoder(r.Body).Decode(&params)
+		if err != nil {
+			httputil.WriteGRPCStyleError(w, codes.Internal, err)
+			return
+		}
```

Comment:
We should also consider adding a max request size to protect the API server (something we should do in all our APIs).  Could be implemented by doing something like:

```go
const maxSize = 100 * 1024 // (100 KiB is way more than needed for this request, ideally make this configurable)

lr := io.LimitReader(r.Body, maxSize)
err := json.NewDecoder(lr).Decode(&params)
if err != nil {
    httputil.WriteGRPCStyleError(w, codes.Internal, errors.Wrap(err, "decoding json request body"))
    return
}
...
```

Also we should add a higher level timeout to the request (also something that is lacking for other APIs):

```go
ctx, cancel := context.WithTimeout(r.Context(), time.Duration(timeout))
defer cancel()
...
```